src/main/kotlin/se/uulm/snowballr/backend/model/SnowballRException.kt

```diff
@@ -175,4 +177,38 @@ sealed class SnowballRException(
     class FailedPreconditionException(
         description: String,
     ) : SnowballRException(description)
+
+    /**
+     * Represents an exception that occurs within the [EmailService].
+     *
+     * @param message The message describing the email-related error.
+     * @param cause The cause of the exception, which can be another exception, or null
+     */
+    sealed class EmailException(
+        message: String,
+        cause: Throwable? = null,
+    ) : SnowballRException(message, cause) {
+        /**
+         * Thrown when an email template file cannot be found or compiled during application startup.
+         * This is a fatal startup error.
+         *
+         * @param templateFileName The name of the file that failed to compile.
+         * @param cause The original IOException from the template engine.
+         */
+        class TemplateCompilationFailed(
+            templateFileName: String,
+            cause: Throwable?,
```

Comment:
If this is always an `IOException` we can use the type directly.
```suggestion
         * @param cause The original [IOException] from the template engine.
         */
        class TemplateCompilationFailed(
            templateFileName: String,
            cause: IOException,
```

Reply:
You're right. This will always be an `IOException`.

Reply:
Also update the comment.